ivadomed/scripts/prepare_dataset_vertebral_labeling.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+##############################################################
+#
+# This script is used to create a dataset with (i) mid-sagittal image and
+# (ii) heatmap of disc labels associated with the mid-sagittal image.
+#
+# Usage: python scripts/automate_training.py -p bids_path -s T2w -a -1
+#
+##############################################################
+
+
+import argparse
+import ivadomed.utils as imed_utils
+import ivadomed.preprocessing as imed_preprocessing
+import nibabel as nib
+import numpy as np
+import os
+import scipy
+
+
+def gaussian_kernel(kernlen=10):
+    """
+    Create a 2D gaussian kernel with user-defined size.
+
+    Args:
+        kernlen (int): size of kernel
+
+    Returns:
+        ndarray: a 2D array of size (kernlen,kernlen)
+    """
+
+    x = np.linspace(-1, 1, kernlen + 1)
+    kern1d = np.diff(scipy.stats.norm.cdf(x))
+    kern2d = np.outer(kern1d, kern1d)
+    return kern2d / kern2d.sum()
+
+
+def heatmap_generation(image, kernel_size):
+    """
+    Generate heatmap from image containing sing voxel label using
+    convolution with gaussian kernel
+    Args:
+        image (ndarray): 2D array containing single voxel label
+        kernel_size (int): size of gaussian kernel
+
+    Returns:
+        ndarray: 2D array heatmap matching the label.
+
+    """
+    kernel = gaussian_kernel(kernel_size)
+    map = scipy.signal.convolve(image, kernel, mode='same')
+    return map
+
+
+def mask2label(path_label, aim=0):
+    """
+    Retrieve points coordinates and value from a label file containing singl voxel label
+    Args:
+        path_label (str): path of nifti image
+        aim (int): -1 will return all points with label between 3 and 30 , any other int > 0  will return
+        only the coordinates of points with label defined by aim.
+
+    Returns:
+        ndarray: array containing the asked point in the format [x,y,z,value] in the RAS orientation.
+
+    """
+    image = nib.load(path_label)
+    image = nib.as_closest_canonical(image)
+    arr = np.array(image.dataobj)
+    list_label_image = []
+    # Arr non zero used since these are single voxel label
+    for i in range(len(arr.nonzero()[0])):
+        x = arr.nonzero()[0][i]
+        y = arr.nonzero()[1][i]
+        z = arr.nonzero()[2][i]
+        # need to check every points
+        if aim == 0:
+            # we don't want to account for pmj (label 49) nor C1/C2 which is hard to distinguish.
+            if arr[x, y, z] < 30 and arr[x, y, z] != 1:
+                list_label_image.append([x, y, z, arr[x, y, z]])
+        elif aim > 0:
+            if arr[x, y, z] == aim:
+                list_label_image.append([x, y, z, arr[x, y, z]])
+    list_label_image.sort(key=lambda x: x[3])
+    return list_label_image
+
+
+def extract_mid_slice_and_convert_coordinates_to_heatmaps(bids_path, suffix, aim=-1):
```

Comment:
how come `bids_path` does not appear on the [generated doc](https://ivadomed.org/en/lr-deep_vertebral_labeling/scripts.html#ivadomed.scripts.prepare_dataset_vertebral_labeling.extract_mid_slice_and_convert_coordinates_to_heatmaps)?

Reply:
I am not sure what you mean as I am seeing it 
![image](https://user-images.githubusercontent.com/22171999/85618403-c189b600-b62e-11ea-95b2-b266497fcd97.png)

Reply:
here: 
https://github.com/ivadomed/ivadomed/blob/0f02ac410f7734822f0a60a1aee783ced8e45d30/ivadomed/scripts/prepare_dataset_vertebral_labeling.py#L88

but not here:
![image](https://user-images.githubusercontent.com/2482071/85640926-23601500-b65b-11ea-9dc6-6bbf54cb3932.png)

Reply:
@lrouhier nevermind, this seems to be a bug (not specific with your code)-- issue opened https://github.com/ivadomed/ivadomed/issues/318

Reply:
Oh right I didn't see that. Now that you say it, all first arguments are missing for the script documentation.